src/px-script/main/dialog/index.js

```diff
@@ -49,91 +39,101 @@ class Dialog {
         this._el.classList.remove("d-flex");
         document.body.classList.remove("dialog-open");
     }
-
 }
 
-const dialog = (() => {
-    const init = () => {
-        const dialogEls = [...document.querySelectorAll(SELECTORS.DIALOG)];
+const init = ids => {
+    const dialogSelector = ids || ids === "" ? getElementsByIds(ids, "dialog") : document.querySelectorAll(SELECTORS.DIALOG);
+    let dialogEls = [];
 
-        window.px._dialogs = window.px._dialogs || [];
+    if (dialogSelector.length) {
+        dialogEls = [...dialogSelector].map(dialog => {
+            const dialogInstance = new Dialog(dialog);
 
-        if (dialogEls.length) {
-            dialogEls.forEach(dialog => window.px._dialogs.push(new Dialog(dialog)));
+            if (_dialogs.length) {
+                _dialogs.forEach((arrayDialog, index) => {
+                    arrayDialog.id === dialogInstance.id ? _dialogs[index] = dialogInstance : _dialogs.push(dialogInstance);
+                });
+            } else {
+                _dialogs.push(dialogInstance);
+            }
 
-            // Init open buttons
-            document.querySelectorAll(SELECTORS.OPEN).forEach(btn => {
-                const id = btn.dataset.dialogOpen;
-                let dialog;
+            return dialogInstance;
+        });
 
-                window.px._dialogs.forEach(d => d.id === id ? dialog = d : null);
+        // Init open buttons
+        document.querySelectorAll(SELECTORS.OPEN).forEach(btn => {
+            const id = btn.dataset.dialogOpen;
+            let dialog;
+
+            _dialogs.forEach(d => d.id === id ? dialog = d : null);
```

Comment:
What is `_dialogs`, and do they come from?

Reply:
The _dialogs array is a collection of all dialog instances on the current page. It is declared in the file.